src/map/unit.cpp

```diff
@@ -2885,6 +2885,12 @@ int32 unit_attack(struct block_list *src,int32 target_id,int32 continuous)
 
 	nullpo_ret(ud = unit_bl2ud(src));
 
+	// Check for special monster random target mode
+	if (src->type == BL_MOB) {
+		mob_data& md = *reinterpret_cast<mob_data*>(src);
```

Comment:
this should now be a `*static_cast<mob_data*>(src);`

Reply:
Done!